### PR TITLE
Get multicast mac

### PR DESCRIPTION
**Pull request by ffr4nz**
For use in fuzzing probes

### CODE REVIEW

**Comment by p-l-** [comment]
Hi,

I'm not sure to understand how this is useful compared to using `RandMAC()` with specific templates. Could you elaborate on that?

File: scapy/volatile.py [hunk 1]
```diff
@@ -274,6 +274,115 @@ def __init__(self, template="*"):
...
..."01:80:C2:00:00:00","09:00:02:04:00:01","09:00:02:04:00:02","09:00:09:00:00:01","09:00:09:00:00:01","09:00:09:00:00:04","09:00:1E:00:00:00","09:00:2B:00:00:00","09:00:2B:00:00:01","09:00:2B:00:00:02","09:00:2B:00:00:03","09:00:2B:00:00:04","09:00:2B:00:00:05","09:00:2B:00:00:06","09:00:2B:00:00:07","09:00:2B:00:00:0F","09:00:2B:00:00:10","09:00:2B:00:00:11","09:00:2B:00:00:12","09:00:2B:00:00:13","09:00:2B:00:00:14","09:00:2B:00:00:15","09:00:2B:00:00:16","09:00:2B:00:00:17","09:00:2B:00:00:18","09:00:2B:00:00:19","09:00:2B:00:00:1A","09:00:2B:00:00:1B","09:00:2B:00:00:1C","09:00:2B:00:00:1D","09:00:2B:00:00:1E","09:00:2B:00:00:1F","09:00:2B:01:00:00","09:00:2B:01:00:01","09:00:2B:02:00:00","09:00:2B:02:01:00","09:00:2B:02:01:01","09:00:2B:02:01:02","09:00:2B:04:00:00","09:00:2B:23:00:00","09:00:4E:00:00:02","09:00:56:00:00:00","09:00:56:FE:FF:FF","09:00:56:FF:00:00","09:00:56:FF:FF:FF","09:00:77:00:00:01","09:00:7C:02:00:05","09:00:7C:05:00:01","0D:1E:15:BA:DD:06","AB:00:00:01:00:00","AB:00:00:02:00:00","AB:00:00:03:00:00","AB:00:00:04:00:00","AB:00:00:05:00:00","AB:00:03:FF:FF:FF","AB:00:03:00:00:00","CF:00:00:00:00:00","09:00:2B:03:xx:xx","AB:00:04:00:xx:xx","AB:00:04:01:xx:yy"]
```
Comment (speakinghedge) [inline scapy/volatile.py]: there are some issues with that kind of lists.
1. the list name misses a context. there are other protocols using the multicast concept too.
2. that kind of list suggest to be an exhaustive collection of addresses (at least for me - but I'm a friend of the damn 'principle of least astonishment' - and I just checked for 01:80:C2:00:00:0E) <=> a complete list needs maintenance, an incomplete list shows limited usage

File: scapy/volatile.py [hunk 2]
```diff
@@ -274,6 +274,115 @@ def __init__(self, template="*"):
...
+	
+	+info: 
+		www.iniqua.com/labs
+	"""
+	def __init__(self, Index=0):
+		self.randMac = None
+		self.indexMac = None
+		self.mac = ()
+		template = "ff:ff:ff:ff:ff:ff"
+		if Index == 0:
```
Comment (speakinghedge) [inline scapy/volatile.py]: those two big blocks (index == 0 | !0) should go to separate methods

File: scapy/volatile.py [hunk 3]
```diff
@@ -274,6 +274,115 @@ def __init__(self, template="*"):
...
+			rn = RandNum(0,66111)
+			if 0 <= rn < 63:
+ 				self.randMac = multicastList[rn]
+			elif 63 <= rn < 319:
+				template = multicastList[63]
```
Comment (speakinghedge) [inline scapy/volatile.py]: if the position of elements in a list is important than usually those elements are somehow special and should be put in a different location. if I would go and add the 01:80:C2:00:00:0E in the head of the list I would gracefully destroy your logic. beside this the code should somehow outline why an element is special, e.g. 
```
is_workday = day_of_week < 6
```
so the next guy working on that must not read the whole method to get an idea what happens on line X.

File: scapy/volatile.py [hunk 4]
```diff
@@ -274,6 +274,115 @@ def __init__(self, template="*"):
...
+						v = int(template[i],16)
+					self.mac += (v,)
+				
+			if 575 <= Index < 66111:
+				template = multicastList[65]
+				relativeIndex = Index - 575
+				template = template.split(":")
+				#macof = hex(relativeIndex).split('x')[1]
+				macof = str("%04X" % relativeIndex)
+				if len(macof) < 4:
```
Comment (speakinghedge) [inline scapy/volatile.py]: that goes so deep :) if-if-if is mostly a good indication that you can move some functionality to a separate method(s).

**Comment by p-l-** [comment]
Please re-open if you answer my question about the interest of this code compared to the existing templates for `RandMAC()`.

**Pull request closed without merging**